### DISCUSSION TITLE
My Early Career Crisis (2014 - 2015) | /en/2018/02/career-crisis/

### DISCUSSION THREAD

**Discussion by yihui** [rbind/yihui#1060] [Comments]
### A painful transition of a fresh PhD from academia to industry, and from selfish open-source to product-oriented open-source

https://yihui.org/en/2018/02/career-crisis/

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
I'm glad to know that you found some of my posts useful. For the reference links, yes, I often have tons of links in my posts (some are quite subtle), so you need to be a little cautious not to be led astray :) When I read Wikipedia pages, I often forget what I was looking for originally after clicking those links to go from one page to another, and yet another...

> Originally posted on 2018-09-26 14:21:19

[a comment by an automated account is not shown]

**Reply by yihui** [reply]
It is a process indeed. Believe it or not, I've been experiencing my second career crisis for a few months and just walked out of my first-ever breakdown, but this time I probably will not write a blog post about it (for complicated reasons). This year has been incredibly tough. The pandemic has so many unexpected consequences. I've been thinking hard about what the pain was trying to remind me of learning. I guess the first lesson to me was to be more tolerant with other people: we could be standing right in each other's blind spot. I got hit in a blind spot really hard this time.

Anyway, in this particularly difficult time, I'm particularly grateful to people like you for your care. All the best!

> Originally posted on 2020-12-05 04:15:09

[a comment by an automated account is not shown]

**Comment by Venkataragavan** [comment]
Hi Yihui, 

I discovered your blog only recently, and it is heartening to read the words you write, especially this post. When I first read through some of them, I could only be intimidated by the sheer amount of work you have done. However, posts like this and a few others, make it more relatable that every one of us started out making dumb mistakes and then the truly efficient ones are those that learn from them. I have a hard time deciding to focus my energy on one project (general projects, not computer-oriented), since several other projects, sometimes unrelated, fight to occupy my curiosity. Your words offer me perspective and hope in resolving problems in a manner that future me would be proud of.

Thank you for writing these words, and being honest with ups and downs in your life. I seem to be on a marathon reading all of your posts.

**Reply by yihui** [reply]
Thank you for the kind words!

> I seem to be on a marathon reading all of your posts.

Yes, I have written too many posts these years, but have slowed down quite a bit since the pandemic. I'm hoping to pick up the marathon in 2023. You should feel lucky if you can't read [Chinese](https://yihui.org/cn/) :)

**Comment by czhang85** [comment]
Hi Yihui, thanks for sharing. I just came back from your most recent post in 2024 and read about your “issues” at the beginning of your career.

I only had two statistics courses in my life but I was surprisingly drawn to your story and even cried when seeing you got hurt, but eventually found the cure. It is interesting to see how people at different stages in their life, sharing different experiences and passions, could find resonance with each other. Perhaps it is because I am also a person that “always has some issues” and ridiculous OCD, and the cause of both is, not unexpected, a share of empathy and intolerance of imperfection. There is unlikely to be a cure once and for all, and I believe the process of solving these issues itself is probably the meaning of life—like how you are into debugging, I find sharing words with the communities an interest of mine.

I was also surprised by your unwavering effort in developing packages, platforms, and resources for the use of R in China and beyond. As a student who once used them in class, I would say that your passion in career as a software engineer has far-reaching impact (I am also a person bad at accepting praises but this one is true from the heart). It is true, as the quote you shared above, “every decision you make, from what you eat to what you do with your time tonight, turns into who you are tomorrow.” In a world as today, many find it impossible to stick to what they like and not yield to the pressure of life. I was moved by your persistence and I truly hope in another post, perhaps ten or twenty years later, you are still on the way to the future you have always dreamt of.

I hope life could “hurt” people with dedicated pursuit like you less so that they could still choose to “believe.”

I like how you end with “the future” section in each and every one of your posts. I was not a straight A student throughout my whole life and managed to work out the right answer for each stage, but words as such make me believe that there is always the possibility, and the need, to search for something new, something unexpected tomorrow. In the end, this is what enchanting about creation, right? Oftentimes I turn into the world of arts, literature, and history for help and a moment of relief. I am not sure whether those texts or images from thousands of years ahead of me would provide any answer into the future.

“Don’t let life randomly kick you into the adult you don’t want to become.” Hope we can all follow these words and answer the heart’s true calling.

I once wrote a retrospective on Cillian Murphy in student newspaper, and there is a quote that I find empowering and would love to share with you: “I can’t remember which director said it, but he said it takes 30 years to make a good actor.”

Who knows how many years it takes to excel in life, but not giving up is the answer with no use-by dates.

**Reply by yihui** [reply]
Hi Cindy,

I read your comments last night and I was so very moved. Since I have been inundated by messages and emails in the past few days, please forgive my brevity.

One major thing I learned from the comments from almost six years ago was that we can gain enormous support from others when we are willing to open our vulnerable heart. It certainly takes courage to (publicly) admit our failure and talk about setbacks, but once you truly believe that no one is perfect, you will have the courage to allow yourself to make mistakes and admit them. You no longer need to be harsh to yourself. In the mean time, you will allow everyone else to make mistakes and it will be much easier to forgive them, too. Frankly speaking, I have never taken this to my heart until last year. I have always been vulnerable (and will continue to be so in the future), and the change last year in me was that nothing could easily hurt me anymore. I have learned to be a little nicer to myself, and lower my expectation of other people. I believe you will never cry again because I get hurt by anything. From now on, you can celebrate on everything, good or bad, that happens to me. (BTW, not sure if you have listened to "[Eyes on me](https://youtu.be/KDGMU6sdVBE)"---I just sang this song this afternoon while doing karaoke at home, and your comments came to my mind.)

> I believe the process of solving these issues itself is probably the meaning of life

I absolutely agree.

> Oftentimes I turn into the world of arts, literature, and history for help and a moment of relief. I am not sure whether those texts or images from thousands of years ahead of me would provide any answer into the future.

Me too. I rarely read books related to my profession. I read a lot of books on the art side instead. You made me think of one of my favorite books, [Van Gogh's Letters](https://www.amazon.com/Van-Goghs-Letters-Paintings-1875-1890/dp/1579128599) (I read the Chinese translation since I'm very slow in reading English). If you liked my post, I think you will love this book as much as I do (you might cry more than once while reading). Many people know van Gogh by his paintings, but I was even more impressed by his letters.

I don't know if anyone else has mentioned it to you before, but hey, I can sense your talent of writing letters like van Gogh. Do you have a personal website? If not, I can help you set up one. I'll be interested in reading more of your writings.

**Reply by czhang85** [reply]
Hi Yihui, when drafting this reply, I am sitting in the corner of a cafe in Florence and, what a coincidence, having “Eyes On Me” playing in my headphones.

Just last week I visited Musée d'Orsay in Paris and saw the Van Gogh exhibition “The Final Moments.” His writings and bold brush strokes, particularly with the bare roots painted on the same day as his suicide, stood out to me. “My life too is attacked at the very root," he wrote. I will probably pick up his letters to read on road as you recommend. It is a pity that I could not read in Dutch to see the world as he did. Van Gogh, among a few other artists I like, have the talent to appreciate solitude.

Thank you for replying promptly in a busy day. I speak less in person but often felt like having a tones of words to spill out in writing, especially when reading thought-provoking posts as yours. Speaking of which, I once spoke to Coco Krumme, an applied mathematician and writer, and I also enjoyed reading her reflections on both professional pursuit and personal life.

It would be wonderful if you could help in setting up a personal website! It has always been my dream to archive my drifting thoughts. Could I send you a book by mail after I returned to the US? I published book reviews sometimes and received some great copies to read last year, and it would be my pleasure to share them with someone I could have great conversations with.

**Reply by yihui** [reply]
I can't believe the coincidence... It's also hard to believe that this is the first time that I have publicly mentioned the Van Gogh letters to someone, and when I mentioned it, you just saw his exhibition. 

Anyway, my address is here: https://github.com/yihui/yihui.org/blob/master/content/en-vitae.md?plain=1#L13 Thank you!

I'll work on the website as soon as I have some free time. It shouldn't take more than a few minutes if you don't mind getting started with a simple theme.

**Reply by czhang85** [reply]
Hi Yihui, a simple theme is more than enough. Follow whichever option works best for you. Much appreciated. Within a few days I will also send a travel bookmark, which I have a hobby of collecting, along with the book. An interesting fact I learned in Italy is that the tradition of collecting souvenirs started with landscape painters such as Antonio Canal. If we were in the eighteenth-century Venice, I would buy an excellent painting instead of a bookmark for you.

**Comment by yongchuang1625** [comment]
Yihui, thank you for sharing your real story. I have experiences similar to yours, and this post is very helpful to me. I am afraid of making mistakes, and when I do make mistakes, I choose to escape them. After reading this post, I started to change step by step. Thank you for your sharing. COS is a very good community to help me learn data science. Thank you for your efforts.
